cmd/main_postsubmit.go

```diff
@@ -1,128 +1,192 @@
 package main
 
 import (
+	"flag"
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"os/exec"
-	"strconv"
+	"path/filepath"
 	"strings"
 )
 
+var (
+	outputStream io.Writer = os.Stdout
+	errStream    io.Writer = os.Stderr
+)
+
 type Command struct {
 	releaseBranch        string
 	gitRoot              string
 	release              string
 	artifactBucket       string
 	uploadToPublicBucket bool
-	args                 []string
+
+	makeTarget string
+	makeArgs   []string
+
+	dryRun bool
 }
 
-func (cmd Command) run(runMake bool, uploadArtifacts bool, targetPath string) {
-	if runMake == true {
-		// Add a specific target
-		cmd.args[1] = cmd.args[1] + targetPath
-		command := exec.Command("make", cmd.args...)
-		output, err := command.CombinedOutput()
+func (c *Command) runMake(uploadArtifacts bool, targetPath string) error {
+	commandArgs := []string{
+		"-C",
+		filepath.Join(c.gitRoot, "projects", targetPath),
+		c.makeTarget,
+	}
+	commandArgs = append(commandArgs, c.makeArgs...)
+
+	cmd := exec.Command("make", c.makeArgs...)
+	log.Printf("Executing: %s", strings.Join(cmd.Args, " "))
+	cmd.Stdout = outputStream
+	cmd.Stderr = errStream
+	if !c.dryRun {
+		err := cmd.Run()
 		if err != nil {
```

Comment:
Probably we would want to print the output when this call succeeds. Same for the create_final_dir.sh script as well.

Reply:
The output of the command is set to forward to the tools stderr/stdout since we don't need to capture/parse it.